sdk/core/Azure.Core.TestFramework/src/ProxyTransport.cs

```diff
@@ -66,6 +66,11 @@ public ProxyTransport(TestProxy proxy, HttpPipelineTransport transport, TestReco
 
         private async Task ProcessAsyncInternalAsync(HttpMessage message, bool async)
         {
+            if (_recording.Mode == RecordedTestMode.Playback && _filter() == EntryRecordModel.DoNotRecord)
```

Comment:
Actually, I'm not sure if this will help as I expect there will be no response on the message so clients will still throw. This check probably needs to be done upstream in the individual tests that need it.

Reply:
We could at least throw here to make the issue easier to diagnose. /cc @heaths